packages/react-components/react-utilities/src/utils/properties.test.ts

```diff
@@ -1,31 +1,35 @@
-import * as React from 'react';
 import { getNativeProps, divProperties } from './properties';
 
 describe('getNativeProps', () => {
   it('can pass through data tags', () => {
-    const result = getNativeProps<React.HTMLAttributes<HTMLDivElement>>(
+    const result = getNativeProps(
       {
         'data-automation-id': 1,
       },
-      divProperties,
+      divProperties as typeof divProperties & {
```

Comment:
because our TS version doesn't support tagged literal as index types we cannot types this properly. 

Do we wanna move forward with this (Breaking change) or we should go back to following:

> **Q:** can we determine if partners use this function in their codebase ? We export it from react-components so it's a public API

```ts
// Implementation with type unaware Allowed Prop Names
// This will not return proper type but will allow/resolve data-* aria-*  properly
declare function getNativeProps<
  Props extends Record<string, unknown>,
  E extends Extract<keyof Props, string> = never
>(
  props: Props,
  allowedPropNames: string[] | Record<string, 1>,
  excludedPropNames?: E[],
): Omit<Props, E>

// Example:
// $ExpectType {a:number, c:number} - actual return value is  {a:number} 🆘 
const result = getNativeProps({ a: 1, b: 2, c: 3 }, ['a', 'b'], ['b']);

```